### ISSUE TITLE
How to resume training

### ISSUE THREAD

**Issue by binarybehemoth** [automl/auto-sklearn#549]
I have retained the tmp folders and set shared_mode to True.

Now, should I just pickle the auto-sklearn object and call fit( ) the second time? Should I set a different seed? Will refit( ) produce an ensemble that includes all the models produced by the different seeds of the instances run at different times and stored in the models/ folder?

What are the necessary steps precisely?

**Comment by mfeurer** [comment]
> Now, should I just pickle-load the auto-sklearn object and call fit( ) the second time?

I should be sufficient to create a new object pointing to the same directory and using `shared_mode = True`. 

> Should I set a different seed?

You must do so.

> Will refit( ) produce an ensemble that includes all the models produced by the different seeds of the instances run at different times and stored in the models/ folder?

Yes.

> What are the necessary steps precisely?

Please check [this unittest](https://github.com/automl/auto-sklearn/blob/master/test/test_automl/test_estimators.py#L88).

If you succeed, it would be awesome if you could contribute your script as an example.

**Comment by binarybehemoth** [comment]
ok! 
First let me try it out...

**Comment by binarybehemoth** [comment]
For subsequent calls to fit( ), should I also set:
                initial_configurations_via_metalearning = 0
                smac_scenario_args = {'initial_incumbent': 'RANDOM'}

?

**Comment by mfeurer** [comment]
Yes. It might be detected that these are known configurations, but I'm not sure about this.

**Comment by binarybehemoth** [comment]
In my first call to fit(), I obtained models A,B and,C. In my second call, I obtained the models D and E but the metric remained unchanged surprisingly. Should I also call fit_ensemble( ) to merge all the models?

A and D are dummies.

**Comment by mfeurer** [comment]
Yes, you need to build an ensemble in the end by using `fit_ensemble` or by having `ensemble_size > 1` for the last call to Auto-sklearn.

**Comment by binarybehemoth** [comment]
Hmmm... as my training proceeds, it seems that the best predictive power is obtained by the first model. Adding subsequent models (up to 4) to the ensemble continued to worsen the predictive power. Is this normal?

**Comment by mfeurer** [comment]
Ensemble building is a greedy procedure. There's no guarantee that it will improve or keep the same performance as new models are added. Keep in mind that the first model can be quite strong depending on the meta-learning.
